lib/mapObjects/CGTownInstance.cpp

```diff
@@ -570,6 +568,17 @@ GrowthInfo CGTownInstance::getGrowthInfo(int level) const
 	return ret;
 }
 
+int CGTownInstance::getDwellingBonus(const std::vector<CreatureID>& creatureIds, const std::vector<ConstTransitivePtr<CGDwelling> >& dwellings) const
+{
+	return std::accumulate(dwellings.cbegin(), dwellings.cend(), 0,
+			[&creatureIds] (int count, decltype(dwellings[0])& dwelling) -> int {
```

Comment:
1) too long statement, please declare at least outer lambda as variable
2) fix formatting https://wiki.vcmi.eu/Coding_guidelines

Reply:
1. I simplified the code by replacing the lambda with `for` loops. Let's make the code more readable.
2.  I read the coding style guidelines on the wiki, however, I don't follow why putting space in-between `if` (or `for`) and `(` is a bad style? Both Mono-project (which is the main source the wiki refers to) and [Google C++ coding style](https://google.github.io/styleguide/cppguide.html#Conditionals) list it as good.
3. Can we add some automatic code style checking script upon pull request? I'd be happy to do some investigation on this.

Reply:
1. Thanks
2. We have our guidelines. We open for discussion of improving them.
3. Planned, see #347